### PR TITLE
check broker user for old version of authenticator.

### PR DESCRIPTION
**Pull request by omercs**
Issue #396
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/weijjia%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1166233%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/399%23issuecomment-123512555%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-07-22T00%3A00%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1166233%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/weijjia%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20b96f73ff1e27cf53461c1232dcdd9688311aaa7d%20src/src/com/microsoft/aad/adal/BrokerProxy.java%2073%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/399%23discussion_r35164169%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20only%20compare%20the%20username%20with%20the%20first%20item%20in%20the%20account%20list%3F%22%2C%20%22created_at%22%3A%20%222015-07-21T22%3A51%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1166233%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/weijjia%22%7D%7D%2C%20%7B%22body%22%3A%20%22This%20version%20of%20authenticator%20has%20one%20user.%22%2C%20%22created_at%22%3A%20%222015-07-21T22%3A53%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/src/com/microsoft/aad/adal/BrokerProxy.java%3AL510-543%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/399%23issuecomment-123512555%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/399%23discussion_r35164169%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-android/pull/399%23discussion_r35164288%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/weijjia'></a>

- [ ] <a href='#crh-comment-Pull b96f73ff1e27cf53461c1232dcdd9688311aaa7d src/src/com/microsoft/aad/adal/BrokerProxy.java 73'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/399#discussion_r35164169'>File: src/src/com/microsoft/aad/adal/BrokerProxy.java:L510-543</a></b>
- <a href='https://github.com/weijjia'></a> why only compare the username with the first item in the account list?
- <a href='https://github.com/omercs'></a> This version of authenticator has one user.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/399?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/399?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-android/pull/399?approve=0'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/399'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by RandalliLama** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/399'>Refresh</a>.<a href='#crh-update-none'></a>

File: src/src/com/microsoft/aad/adal/BrokerProxy.java [hunk 1]
```diff
@@ -498,6 +510,33 @@ private boolean checkAccount(final AccountManager am) {
         return false;
     }
 
+    private boolean verifyAccount(Account[] accountList, String username, String uniqueId) {
+        if (!StringExtensions.IsNullOrBlank(username)) {
+            return username.equalsIgnoreCase(accountList[0].name);
```
Comment (weijjia) [inline src/src/com/microsoft/aad/adal/BrokerProxy.java]: why only compare the username with the first item in the account list?

Comment (omercs) [inline src/src/com/microsoft/aad/adal/BrokerProxy.java]: This version of authenticator has one user.

**Comment by weijjia** [comment]
:shipit:

**Pull request merged**